python/orca/src/bigdl/orca/learn/pytorch/callbacks.py

```diff
@@ -0,0 +1,101 @@
+        for this method but may change in the future.
+        """
+        pass
+
+    @abstractmethod
+    def on_batch_end(self, batch, logs=None):
+        """
+        A backwards compatibility alias for `on_train_batch_end`.
+        @param batch: Integer, index of batch within the current epoch.
+        @param logs:  Dict. Aggregated metric results up until this batch.
+        """
+        pass
+
+    @abstractmethod
+    def on_epoch_begin(self, epoch, logs=None):
+        """
+        Called at the start of an epoch.
+        Subclasses should override for any actions to run. This function should only
+        be called during TRAIN mode.
+        @param epoch: Integer, index of epoch.
+        @param logs: Dict. Currently, saved stats in last epoch has been passed to this argument
+        for this method but may change in the future.
+        """
+        pass
+
+    @abstractmethod
+    def on_epoch_end(self, epoch, logs=None):
+        """
+        Called at the end of an epoch.
+        Subclasses should override for any actions to run. This function should only
+        be called during TRAIN mode.
+        @param epoch:  Integer, index of epoch.
+        @param logs: Dict, metric results for this training epoch.
+        """
+        pass
+
+    @abstractmethod
+    def on_train_batch_begin(self, batch, logs=None):
+        """
+        Called at the beginning of a training batch in `fit` methods.
+        Subclasses should override for any actions to run.
+        @param batch: Integer, index of batch within the current epoch.
+        @param logs: Dict. Currently, no data is passed to this argument for this method
+          but that may change in the future.
+        """
+        pass
+
+    @abstractmethod
+    def on_train_batch_end(self, batch, logs=None):
+        """
+        Called at the end of a training batch in `fit` methods.
+        Subclasses should override for any actions to run.
+        @param batch: Integer, index of batch within the current epoch.
+        @param logs:  Dict. Aggregated metric results up until this batch.
+        """
+        # For backwards compatibility.
+        self.on_batch_end(batch, logs=logs)
+        pass
+
+    @abstractmethod
+    def on_train_begin(self, logs=None):
+        """
+        Called at the beginning of training.
+        Subclasses should override for any actions to run.
+        @param logs: Dict. Currently, no data is passed to this argument for this method
+          but that may change in the future.
+        """
+        pass
+
+    @abstractmethod
+    def on_train_end(self, logs=None):
+        """
+        Called at the end of training.
+        Subclasses should override for any actions to run.
+        @param: Dict. Currently, the output of the last call to `on_epoch_end()`
+          is passed to this argument for this method but that may change in
+          the future.
+        """
+        pass
+
+    @abstractmethod
+    def set_model(self, model):
```

Comment:
Do we need to make this one an `abstractmethod`? I do not think a user should override this?